auto_round/mllm/mllm_dataset.py

```diff
@@ -129,10 +129,23 @@ def vlm_encode(sources, template: "Template"):
     return element
 
 
+def _extract_data_dir(dir_path):
+    if os.path.isdir(dir_path):
+        return dir_path
+    else:
+        result = {}
+        dir_path = dir_path.split(",")
+        for _path in dir_path:
+            k, v = _path.split('=')
+            if k in ['image', 'video', 'audio']:
+                result[k] = v
+        return result
+            
+
 class LlavaDataset(Dataset):
```

Comment:
support register, we may use other dataset. Also support customized dataset for users which allow users to pass a json which includes text or chat text and image path as the dataset